### PR TITLE
restrict numsymb access to `symbolic-operator` fn

### PR DESCRIPTION
**Pull request by sritchie**
This allows us to make the interface fully closed, and properly switch between the "constructor simplification" world and the current implementations when we add those.

### CODE REVIEW

File: src/sicmutils/polynomial.cljc [hunk 1]
```diff
@@ -474,17 +474,20 @@
...
-           (sort-by exponents #(monomial-order %2 %1))
-           (map (fn [[xs c]]
-                  (->> (map (fn [exponent var]
-                              (sym/expt var exponent))
-                            xs vars)
-                       (reduce sym/mul 1)
-                       (sym/mul c))))
-           (reduce sym/add 0))
-      p))
+    (let [*    (sym/symbolic-operator '*)
```
Comment (sritchie) [inline src/sicmutils/polynomial.cljc:477]: @littleredcomputer , this is the flavor of change I pushed through. Let me know what you think here.

File: src/sicmutils/simplify.cljc [hunk 2]
```diff
@@ -18,19 +18,16 @@
 ;
 
 (ns sicmutils.simplify
-  (:require [clojure.walk :refer [postwalk]]
```
Comment (sritchie) [inline src/sicmutils/simplify.cljc:21]: removed extra imports here.

File: src/sicmutils/polynomial.cljc (hunk 1 above)
Comment (littleredcomputer) [reply src/sicmutils/polynomial.cljc:477]: It seems to work out to the same thing, more or less? The definition of symbolic-operator knows the definition of sym/add ...

**Review by littleredcomputer — APPROVED** [review]

File: src/sicmutils/polynomial.cljc (hunk 1 above)
Comment (sritchie) [reply src/sicmutils/polynomial.cljc:477]: Yes, for now it does. I want to add (for completeness, oh how I'm regretting thinking this way) the ability to toggle the more-advanced constructor simplifications from `numsymb.scm` on and off with a dynamic variable. This one step of indirection makes that possible.

**Pull request merged**